content/en/blog/_posts/2023-01-17-confidential-kubernetes.md

```diff
@@ -0,0 +1,155 @@
+
+Fully (also partially and somewhat) homomorphic encryption allows one to perform computations, such as addition or multiplication, on encrypted data. This provides the property of encryption in use but does not provide integrity protection or attestation.
+
+Zero Knowledge Proofs or Protocols are a privacy-preserving technique (PPT) that allows one party to prove facts about their data without revealing anything else about the data. ZKP can be used instead of or in addition to confidential computing to protect the privacy of the involved parties and their data. Similarly, Multi-Party Computation enables multiple parties to work together on a computation, i.e, each party provides their data to the result, without leaking their data to any of the other parties.
+
+# Use Cases of Confidential Computing
+
+The presented confidential computing platforms show that both the isolation of a single process and therefore minimization of the trusted computing base, and the isolation of a full virtual machine are possible. This already enabled a lot of interesting and secure projects to emerge:
+
+## Confidential Containers
+
+The [Confidential Containers project](https://github.com/confidential-containers) enables users to run a container inside a confidential context. It provides an abstraction layer so that users do not need to interface directly with the confidential computing hardware.
+
+Confidential Containers is a [CNCF sandbox project](https://www.redhat.com/en/blog/what-confidential-containers-project).
+
+> TODO: @Tobin/Mikko (or other CoCo folks) can you expand on this chapter?
+
+## Managed Confidential Kubernetes
+
+[Azure](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-node-pool-aks) and [GCP](https://cloud.google.com/blog/products/identity-security/announcing-general-availability-of-confidential-gke-nodes) both support the use of confidential virtual machines as worker nodes for their managed Kubernetes offerings.
+
+Both services aim for better workload protection and security guarantees by enabling memory encryption for container workloads. However, they don't seek to fully isolate the cluster or workloads against the service provider or infrastructure. Specifically, they don't offer a dedicated confidential control plane or expose attestation capabilities for the confidential cluster/nodes.
+
+Azure also enables [Confidential Containers](https://github.com/confidential-containers) in their managed Kubernetes offering. They support the creation based on [Intel SGX enclaves](https://learn.microsoft.com/en-us/azure/confidential-computing/confidential-containers-enclaves) and [AMD SEV-based VMs](https://techcommunity.microsoft.com/t5/azure-confidential-computing/microsoft-introduces-preview-of-confidential-containers-on-azure/ba-p/3410394).
```

Comment:
Can we list other providers, eg Alibaba - https://www.alibabacloud.com/help/en/container-service-for-kubernetes/latest/tee-based-confidential-computing-tee-based-confidential-computing?

We should also assert what the trust model / threat model looks like for a provider. For example, if there is no attestation capability, do you need to trust that your provider is delivering the service they claim to? Also, do we now rely on the CPU vendor more than we previously did?

Reply:
from what I can tell Alibaba's ACK is using inclavare and occulum.  which is great! but we should probably list Inclavare and Occulum at the top level, and examples of *using* those as use cases.  Same for Azure too I think.